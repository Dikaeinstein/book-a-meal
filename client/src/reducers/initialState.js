const initialState = {
  urls: [
    {
      id: 1,
      name: 'Signin',
      link: 'signin',
    },
    {
      id: 2,
      name: 'Signout',
      link: 'signout',
    },
  ],
  menus: {
    isFetching: false,
    data: {
      menu: {
        name: 'Menu for today',
        meals: [],
      },
      menus: [],
    },
    isSet: false,
    isSaving: false,
    fetchError: null,
    saveError: null,
    updateError: null,
    isUpdating: false,
  },
  user: {
    isSubmitting: false,
    loggedIn: false,
    data: {
      id: null,
      name: '',
      email: '',
      role: '',
    },
    error: null,
  },
  orders: {
    isFetching: false,
    isSaving: false,
    isUpdating: false,
    isDeleting: false,
    data: {
      userOrders: [],
      checkedOutOrder: {},
    },
    fetchError: null,
    saveError: null,
    updateError: null,
    deleteError: null,
  },
  meals: {
    isFetching: false,
    isUpdating: false,
    isSaving: false,
    data: [],
    fetchError: null,
    saveError: null,
    updateError: null,
  },
  dashboard: {
    isFetchingTotalOrders: false,
    isFetchingTotalSales: false,
    isUpdating: false,
    isDeleting: false,
    fetchTotalOrdersError: null,
    fetchTotalSalesError: null,
    isFetchingAllOrders: false,
    fetchAllOrdersError: null,
    deleteOrderError: null,
    updateOrderError: null,
    totalAmount: 0,
    totalOrders: 0,
    allOrders: [],
  },
};

export default initialState;
