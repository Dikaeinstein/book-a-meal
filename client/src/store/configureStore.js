import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
