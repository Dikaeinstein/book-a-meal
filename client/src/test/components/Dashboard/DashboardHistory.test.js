import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedDashboardOrderHistory, { DashboardOrderHistory }
  from '../../../components/Dashboard/DashboardOrderHistory';
import initialState from '../../../reducers/initialState';

describe('<DashboardHistory />', () => {
  const order = {
    id: 11,
    quantity: 1,
    total: '4000',
    status: 'delivered',
    expired: false,
    mealId: 3,
    userId: 4,
    meal: {
      id: 3,
      name: 'Amala and Ewedu',
      price: '4000',
    },
    user: {
      id: 4,
      name: 'Chinonso Okorie',
    },
  };
  const props = {
    isFetching: false,
    getAllOrders: jest.fn(),
    getCatererOrders: jest.fn(),
    allOrders: [order],
    userRole: 'superAdmin',
    superAdminCurrentUrl: 'http://testurl',
    catererCurrentUrl: 'http://testurl',
  };
  it('should render with correct props', () => {
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const wrapper = shallow(<ConnectedDashboardOrderHistory
      {...props}
      store={store}
    />);
    expect(wrapper.props().isFetching).toBeFalsy();
  });
  it('should match snapshot', () => {
    let tree = shallow(<DashboardOrderHistory {...props} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<DashboardOrderHistory
      {...{ ...props, userRole: 'caterer' }}
    />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<DashboardOrderHistory
      {...{ ...props, isFetching: true }}
    />);
    expect(tree).toMatchSnapshot();
  });
});
