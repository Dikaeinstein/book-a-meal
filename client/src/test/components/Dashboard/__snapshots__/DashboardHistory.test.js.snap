// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHistory /> should match snapshot 1`] = `
<section>
  <h5>
    Order history
  </h5>
  <OrderList
    orders={
      Array [
        Object {
          "expired": false,
          "id": 11,
          "meal": Object {
            "id": 3,
            "name": "Amala and Ewedu",
            "price": "4000",
          },
          "mealId": 3,
          "quantity": 1,
          "status": "delivered",
          "total": "4000",
          "user": Object {
            "id": 4,
            "name": "Chinonso Okorie",
          },
          "userId": 4,
        },
      ]
    }
  />
  <Paginate
    nextUrl=""
    onPageChange={[Function]}
    previousUrl=""
    style={
      Object {
        "marginTop": "0",
      }
    }
  />
</section>
`;

exports[`<DashboardHistory /> should match snapshot 2`] = `
<section>
  <h5>
    Order history
  </h5>
  <OrderList
    orders={
      Array [
        Object {
          "expired": false,
          "id": 11,
          "meal": Object {
            "id": 3,
            "name": "Amala and Ewedu",
            "price": "4000",
          },
          "mealId": 3,
          "quantity": 1,
          "status": "delivered",
          "total": "4000",
          "user": Object {
            "id": 4,
            "name": "Chinonso Okorie",
          },
          "userId": 4,
        },
      ]
    }
  />
  <Paginate
    nextUrl=""
    onPageChange={[Function]}
    previousUrl=""
    style={
      Object {
        "marginTop": "0",
      }
    }
  />
</section>
`;

exports[`<DashboardHistory /> should match snapshot 3`] = `
<div
  className="loader-container"
>
  <Loading
    text="fetching orders..."
  >
    <Preloader
      active={true}
      color="blue"
      flashing={true}
      size="medium"
    />
  </Loading>
</div>
`;
